CRM/Core/DAO.php

```diff
@@ -3007,9 +3007,28 @@ public static function getSelectWhereClause($tableAlias = NULL) {
     return $clauses;
   }
 
+  /**
+   * Check whether the given action can be performed on these values.
+   *
+   * The return is the input values array with inaccessible items filtered out.
+   *
+   * @param string $action
+   * @param array $values
+   * @param null $contactID
+   *
+   * @return array
+   */
+  public function checkAccess(string $action, array $values, $contactID = NULL): array {
+    if (is_null($contactID)) {
+      $contactID = CRM_Core_Session::getLoggedInContactID();
+    }
+    CRM_Utils_Hook::checkAccess(CRM_Core_DAO_AllCoreTables::getBriefName(get_class($this)), $action, $contactID, $values);
```

Comment:
@eileenmcnaughton I'm guessing we may want to call core methods that already exist before we get to the hook invocation and would we want extensions to potentially override core functions e.g. result of `CRM_Mailing_BAO_Mailing::acl`  or whatever is that function?

Reply:
@seamuslee001 yeah - this is the function on the DAO class which would be overridden by various BAO classes

Reply:
@eileenmcnaughton I guess in my mind at least wouldn't that mean we end up with a lot of duplicated code around the place because you would want to copy n paste a fair chunk of this function on the other DAOs?